src/Lecture1.hs

```diff
@@ -98,7 +99,16 @@ start position can be considered as zero (e.g. substring from the
 first character) and negative end position should result in an empty
 string.
 -}
-subString start end str = error "TODO"
+subString :: Int -> Int -> [Char] -> [Char]
+subString start end str
+    | end < 0   = ""
+    | start < 0 = take (end + 1) str
+    | otherwise = take (end - start + 1) (drop start str)
```

Comment:
If `start > end`, the function will work correctly because `take` with a non-positive int returns an empty list. And due to Haskell laziness, this will be efficient.

But for clarity reasons, I would add this pre-condition separately 🙂